### PR TITLE
20 add simple entity api

### PR DESCRIPTION
**Pull request by panositi**
endpoints changed to use plural.

### CODE REVIEW

**Review by scp93ch — DISMISSED** [review]
We need tests for these methods.
The JavaDoc on the methods should be completed (and should describe the format of e.g. the "uri" parameter).

File: src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/EntityController.java [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+
+        } catch (Exception e) {
+            logger.error("Simple API get threat failed due to an error", e);
+            throw new InternalServerErrorException(
+                    "Threat fetch failed. Please contact support for further assistance.");
+        }
+    }
+
+    /**
+     * This REST method ...
```
Comment (kenmeacham) [inline src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/EntityController.java:118]: "This REST method" does what?

Comment applies to all methods

**Review by kenmeacham — CHANGES_REQUESTED** [review]
Some general comments:

1) All methods should properly describe what they do. Most currently say "This REST method ..."
2) It is not clear how an entity URI should be defined within the request URI, i.e. the hash should be escaped somehow. Needs to be in the docs somewhere.
3) If I enter an invalid URI fragment, e.g. system#123, this hasn't been escaped, so SSM thinks we are entering "system". The returned content is weird. This suggests we need some validation of the incoming URI fragment
4) Related to the above, it would help if the query result was checked for the correct type, i.e. when getting a threat with a specific URI, then assert that type is "core#Threat". You could also assert that the returned "uri" matches the incoming one
5) The API endpoints for getting levels are a bit confusing. It refers to metrics, but not that we want levels. For example, change "entity/domain/{metric}" to "entity/domain/levels/{metric}".
6) There are no unit tests yet (I will investigate this initially)

**Comment by kenmeacham** [comment]
One more thing. We have a way to query system TWAS, but not yet the domain TWA. This would help with the work we are doing on SSM Adaptor.

File: src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/EntityController.java (hunk 1 above)
Comment (panositi) [reply src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/EntityController.java]: I've added short descriptions for all methods address @kenmeacham comments 1,2,3, and 5

**Comment by panositi** [comment]
> One more thing. We have a way to query system TWAS, but not yet the domain TWA. This would help with the work we are doing on SSM Adaptor.

There is `/models/{modelId}/entity/domain/misbehaviours/{uri}`

**Comment by kenmeacham** [comment]
> > One more thing. We have a way to query system TWAS, but not yet the domain TWA. This would help with the work we are doing on SSM Adaptor.
> 
> There is `/models/{modelId}/entity/domain/misbehaviours/{uri}`

OK, but how does that help? I was asking for available trustworthiness attributes defined in the domain model.

**Review by kenmeacham — APPROVED** [review]

**Comment by kenmeacham** [comment]
Many issues have been addressed. Need to merge into dev and move on.

**Pull request merged**